opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java

```diff
@@ -110,29 +117,35 @@ public void close() {
     }
 
     @Override
-    public synchronized Span setTag(String key, String value) {
-        this.tags.put(key, value);
-        return this;
+    public Span setTag(String key, String value) {
+        return setObjectTag(key, value);
     }
 
     @Override
-    public synchronized Span setTag(String key, boolean value) {
-        this.tags.put(key, value);
-        return this;
+    public Span setTag(String key, boolean value) {
+        return setObjectTag(key, value);
     }
 
     @Override
-    public synchronized Span setTag(String key, Number value) {
-        this.tags.put(key, value);
+    public Span setTag(String key, Number value) {
+        return setObjectTag(key, value);
+    }
+
+    private synchronized Span setObjectTag(String key, Object value) {
+        finishedCheck(String.format("Adding tag {%s:%s} to already finished span", key, value.toString()));
+        tags.put(key, value);
         return this;
     }
 
     @Override
     public final Span log(Map<String, ?> fields) {
         return log(nowMicros(), fields);
     }
+
     @Override
-    public final Span log(long timestampMicros, Map<String, ?> fields) {
+    public final synchronized Span log(long timestampMicros, Map<String, ?> fields) {
+        finishedCheck(String.format("Adding logs %s at %d to already finished span", fields.toString(),
```

Comment:
small nit: while the Mock span is not performance-critical, it would be more efficient (and more DRY) to just pass a variadic list of args to finishedCheck, and only invoke String.format() within that function if the condition is triggered.